lib/process/watch.go

```diff
@@ -18,23 +18,98 @@ package process
 
 import (
 	"context"
+	"crypto/x509"
+	"encoding/pem"
+	"strings"
 	"time"
 
 	"github.com/gravitational/gravity/lib/constants"
 	"github.com/gravitational/gravity/lib/defaults"
 	libkube "github.com/gravitational/gravity/lib/kubernetes"
+	"github.com/gravitational/gravity/lib/ops"
+	"github.com/gravitational/gravity/lib/ops/opsservice"
 	"github.com/gravitational/gravity/lib/processconfig"
 	"github.com/gravitational/gravity/lib/storage"
-
+	"github.com/gravitational/gravity/lib/utils"
 	"github.com/gravitational/teleport/lib/service"
 	"github.com/gravitational/trace"
+
 	v1 "k8s.io/api/core/v1"
 	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
 	"k8s.io/apimachinery/pkg/fields"
 	"k8s.io/apimachinery/pkg/watch"
 	"k8s.io/client-go/kubernetes"
 )
 
+// runCertExpirationWatch checks if the default self signed cluster cert is about to expire soon and updates it
+func (p *Process) runCertExpirationWatch(client *kubernetes.Clientset) clusterService {
+	p.Info("Starting self signed certificate expiration watch.")
+
+	return func(ctx context.Context) {
+		ticker := time.NewTicker(time.Hour * 24)
+		defer ticker.Stop()
+		for {
+			err := p.replaceCertIfAboutToExpire(client)
+			if err != nil {
+				p.Errorf("Failed to check for certificate expiration: %v.", trace.DebugReport(err))
```

Comment:
> Also, I'm a bit wary about letting this loop run on each gravity-site replica - it would make more sense to run on just the leader.

Unless I'm missing anything, it will run on leader as implemented (since it's launched via `p.startService`).

Agree about retrying on error.

Reply:
I think with renewing 30 days before expiry, this leaves plenty of attempts at the 24 hour timer, so my 2 cents is we don't need the added complexity of a rapid retry on error.

Although my corner case concern may be a bit different, I wonder if we have any clusters where the elections are unstable, and I wonder if there is a possibility a cluster won't stay the leader for 24 hours.